File: frontend/src/types/lecture.ts
// src/types/lecture.ts
import type { User } from "./users.ts";

export interface GetLecturesParams {
    page?: number;
    limit?: number;
    sort_by?: string;
    order?: 'asc' | 'desc';
    user_id?: string;
    search_name?: string;
    scope?: string;
}

export interface LectureUpdate {
    name?: string;
    lecturer?: string;
    text?: string;
    public?: boolean;
}

export interface Lecture {
    id: string;
    user: User;
    lecturer: string;
    name: string;
    audio_url: string;
    text: string;
    public: boolean;
    created_at: string;
    updated_at: string;
}

export interface LecturesPage {
    items: Lecture[];
    total: number;
    page: number;
    pages: number;
}

export interface ExampleTaskDescription {
    filename: string;
    title: string;
    description: string;
}

File: backend/src/api/schemas/lecture.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, ConfigDict

from src.api.schemas.user import UserRead


class LectureRead(BaseModel):
    id: UUID
    user: UserRead
    name: str
    lecturer: str
    audio_url: str | None
    text: str
    public: bool
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True, extra="ignore")


class LecturesPage(BaseModel):
    items: list[LectureRead]
    total: int
    page: int
    pages: int


class LectureUpdate(BaseModel):
    name: str | None = None
    lecturer: str | None = None
    text: str | None = None
    public: bool | None = None
